sycl/include/sycl/handler.hpp

```diff
@@ -707,16 +709,52 @@ class __SYCL_EXPORT handler {
         *static_cast<T *>(detail::getValueFromDynamicParameter(DynamicParam));
     // Set the arg in the handler as normal
     setArgHelper(ArgIndex, std::move(ArgValue));
+
     // Register the dynamic parameter with the handler for later association
     // with the node being added
     registerDynamicParameter(DynamicParam, ArgIndex);
   }
 
-  // setArgHelper for graph dynamic_work_group_memory
-  void
-  setArgHelper(int ArgIndex,
-               ext::oneapi::experimental::detail::dynamic_work_group_memory_base
-                   &DynWorkGroupBase);
+  template <typename DataT, typename PropertyListT>
```

Comment:
nit: would it make sense to only have declaration here and provide out-of-class definition somewhere in the graph headers?

Reply:
I think that it makes sense to keep this in the handler header, at least for this PR. There is already a precedent to having `setArgHelper()` implementations in the handler for existing graph functionality (e.g. `dynamic_parameter`) and other extensions (e.g. `work_group_memory`). Moving only part of the implementation to the graph headers can be confusing. However, if you think that the handler header is getting too large / full of extension functionality, we can consider possible refactoring options as part of future work.